staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -79,9 +79,27 @@ func newTestCacher(s storage.Interface) (*Cacher, storage.Versioner, error) {
 		Clock:       clock.RealClock{},
 	}
 	cacher, err := NewCacherFromConfig(config)
+
 	return cacher, storage.APIObjectVersioner{}, err
 }
 
+func newTestCacher(s storage.Interface) (*Cacher, storage.Versioner, error) {
+	cacher, versioner, err := newTestCacherWithoutSyncing(s)
+	if err != nil {
+		return nil, versioner, err
+	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.ResilientWatchCacheInitialization) {
```

Comment:
the way you changed the tests that use this method suggests we will always wait for the cacher to be ready before continuing the tests. Why not add the wait logic to this method unconditionally?

Reply:
It was effectively allow running tests in both old and new mode.

Reply:
Since the feature will be enabled straight away and we don't want to rewrite the test, the tests will have to wait for the cache to be initialised anyway, regardless of whether the feature is enabled or disabled.

Reply:
> regardless of whether the feature is enabled or disabled.

I don't understand - if the feature is disabled it's the current state and you don't need to wait now.

Reply:
I was looking at the `TestGetListNonRecursiveCacheBypass` and here is what I have seen:

```
func TestGetListNonRecursiveCacheBypass(t *testing.T) {
	...
	cacher, _, err := newTestCacher(backingStorage)
	...
	if !utilfeature.DefaultFeatureGate.Enabled(features.ResilientWatchCacheInitialization) {
		if err := cacher.ready.wait(context.Background()); err != nil {
			t.Fatalf("unexpected error waiting for the cache to be ready")
		}
	}
```

after checking `newTestCacher` i realised that this method will wait for the cacher to be ready as well:

```
if utilfeature.DefaultFeatureGate.Enabled(features.ResilientWatchCacheInitialization) {
		// The tests assume that Get/GetList/Watch calls shouldn't fail.
		// However, 429 error can now be returned if watchcache is under initialization.
		// To avoid rewriting all tests, we wait for watcache to initialize.
		if err := cacher.ready.wait(context.Background()); err != nil {
			return nil, storage.APIObjectVersioner{}, err
		}
	}
```

So it seems that regardless of whether the feature is enabled or disabled the tests will wait for the cacher to be ready.

Reply:
Some tests - yes. Not all tests.

Reply:
ok, thanks.